lib/puppet/pops/time/timestamp.rb

```diff
@@ -33,10 +33,13 @@ def self.parse(str, format = DEFAULT_FORMATS)
     end
   end
 
-  undef_method :-@, :+@, :%,:modulo,:divmod, :div, :fdiv, :abs, :abs2, :magnitude # does not make sense on a Timestamp
+  undef_method :-@, :+@, :div, :fdiv, :abs, :abs2, :magnitude # does not make sense on a Timestamp
   if method_defined?(:negative?)
     undef_method :negative?, :positive?
   end
+  if method_defined?(:%)
+    undef_method :%, :modulo, :divmod
```

Comment:
Ok. I thought, better safe than sorry. It doesn't matter, really. We never
expose those methods.

On Tue, Aug 30, 2016 at 10:13 PM, Patrick Carlisle <notifications@github.com
> wrote:

> In lib/puppet/pops/time/timestamp.rb
> <https://github.com/puppetlabs/puppet/pull/5246#discussion_r76870189>:
>
> >    if method_defined?(:negative?)
> >      undef_method :negative?, :positive?
> >    end
> > +  if method_defined?(:%)
> > +    undef_method :%, :modulo, :divmod
>
> whether modulo and divmod are defined doesn't seem tied to %. My jruby
> Numeric has both of them but not %.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/5246/files/4f72feaeb6ce78833927e65eac2623531bdc861b..a414067c3fbe811d63a75abe8c300c4ca7f4ae15#r76870189>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMyV-_HKWPMbz6FxaqtyOBZnzGMXKg1ks5qlI72gaJpZM4Jw6L2>
> .
>